tools/rostest/src/rostest/runner.py

```diff
@@ -121,7 +122,10 @@ def fn(self):
 
             #setup the test
             # - we pass in the output test_file name so we can scrape it
-            test_file = xmlResultsFile(test_pkg, test_name, False)
+            env = None
+            if results_base_dir:
+                env = {ROS_TEST_RESULTS_DIR: results_base_dir}
```

Comment:
Other code uses the same pattern: https://github.com/ros/ros_comm/blob/44a089aef8f90243084045db6ad8fe6b95f53052/tools/rostest/src/rostest/rostest_main.py#L148-L154

And at the end of the call stack only this value is checked: https://github.com/ros-infrastructure/rospkg/blob/d3c12bdfbf990b9910220642bafef16b4c830b63/src/rospkg/environment.py#L162-L166